types/static-kdtree/index.d.ts

```diff
@@ -0,0 +1,41 @@
+// Type definitions for static-kdtree 1.0
+// Project: https://github.com/mikolalysenko/static-kdtree
+// Definitions by: Zheyang Song <https://github.com/ZheyangSong>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.7
+
+import ndarray = require("ndarray");
+
+// ref: https://github.com/microsoft/TypeScript/pull/40002
+type TupleOf<T, N extends number> = N extends N ? number extends N ? T[] : _TupleOf<T, N, []> : never;
+type _TupleOf<T, N extends number, R extends unknown[]> = R['length'] extends N ? R : _TupleOf<T, N, [T, ...R]>;
+
+type ISerialized<D extends number = number> = {
+  p: Array<TupleOf<number, D>>;
+  i: number[];
+} | {
+  d: number;
+};
+
+type TVisitor = (id: number) => any;
+
+declare class KDTree<D extends number = number> {
+  readonly dimension: number;
+  readonly length: number;
+
+  constructor(points: ReturnType<typeof ndarray>, ids: number[], n: number, d: D);
+
+  range(lo: number[], hi: number[], visit: TVisitor): number;
+  rnn(point: TupleOf<number, D>, radius: number, visit: TVisitor): void;
+  nn(point: TupleOf<number, D>, maxDistance?: number): number;
+  knn(point: TupleOf<number, D>, maxPoints?: number, maxDistance?: number): number[];
+  serialize(): ISerialized<D>;
+  dispose(): void;
+}
+
+declare function createKDTree<D extends number = number>(points: ReadonlyArray<TupleOf<number, D>>): KDTree<D>;
```

Comment:
are the `options` just tuple or it can be an instance of `ndarray` (which fails `Array.isArray` check)?:
https://github.com/mikolalysenko/static-kdtree#var-kdt--createkdtreepoints

Reply:
> are the `options` just tuple or it can be an instance of `ndarray` (which fails `Array.isArray` check)?: https://github.com/mikolalysenko/static-kdtree#var-kdt--createkdtreepoints

good catch. Thanks.